core/src/main/scala/org/apache/spark/SparkContext.scala

```diff
@@ -1912,6 +1922,17 @@ class SparkContext(config: SparkConf) extends Logging with ExecutorAllocationCli
     // the cluster manager to get an application ID (in case the cluster manager provides one).
     listenerBus.post(SparkListenerApplicationStart(appName, Some(applicationId),
       startTime, sparkUser, applicationAttemptId))
+    val logPrefix = "spark.driver.log."
+    logUrls = Option(
+      System.getProperties.stringPropertyNames()
+        .filter(_.startsWith(logPrefix))
+        .map(key => (key.substring(logPrefix.length), System.getProperty(key)))
+        .toMap
+    )
+    _logUrls.foreach { logUrlsMap =>
+      listenerBus.post(SparkListenerExecutorAdded(System.currentTimeMillis(),
```

Comment:
Hmm, as far as I can see in the current master, `SparkListenerExecutorAdded` was never posted for the driver before. That may have some side-effects that need to be checked.

For example, I'm pretty sure `ExecutorAllocationManager.scala` needs to be changed to ignore the driver.